configs/example/gem5_library/x86-gapbs-benchmarks.py

```diff
@@ -185,30 +165,7 @@
 # committed instructions till ROI ends (marked by `workend`). We then finish
 # executing the rest of the benchmark.
 
-# GAPBS benchmarks can be run using a synthetic graph
-
-if args.synthetic == "1":
-    if args.size == "USA-road-d.NY.gr":
-        print(
-            "fatal: cannot use a real graph with --synthetic 1",
-            file=sys.stderr,
-        )
-        exit(-1)
-
-    command = f"./{args.benchmark} -g {args.size}\n"
-else:
-    command = f"./{args.benchmark} -sf ../{args.size}"
-
-board.set_kernel_disk_workload(
-    # The x86 linux kernel will be automatically downloaded to the
-    # `~/.cache/gem5` directory if not already present.
-    # gapbs benchamarks was tested with kernel version 4.19.83
-    kernel=obtain_resource("x86-linux-kernel-4.19.83"),
-    # The x86-gapbs image will be automatically downloaded to the
-    # `~/.cache/gem5` directory if not already present.
-    disk_image=obtain_resource("x86-gapbs"),
-    readfile_contents=command,
-)
+board.set_workload(benchmark_choices[args.benchmark])
```

Comment:
Then here:
```suggestion
board.set_workload(obtain_resource(args.benchmark))
```

Reply:
We dont need to obtain the resource again as we are already getting it when the suite is initialized as declared in line 113-114.

Reply:
I standby my initial suggestion here. I think it's a lot clearer this way.